DataDog/datadog-agent
# [system-probe] Add HTTP monitoring userspace side

### What does this PR do?

Adds the userspace side of HTTP monitoring in the system probe. The added components are responsible for processing HTTP transactions coming from ebpf, storing http transaction information in the appropriate ConnectionStats objects, and encoding the http data in order to be sent to the process agent & then along to the backend.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan
In the agent VM:
1) Enable HTTP monitoring in `system_probe.yaml`
    ```
    network_config:
          enable_http_monitoring: true
    ```
2) Build & run the system probe
    ```
    inv -e system-probe.build --bundle-ebpf && sudo ./bin/system-probe/system-probe
    ```
3) Make some http requests (must be http, not https)
     - a good endpoint to make requests to is `http://httpbin.org/status/{code}`, which generates a response with the given code
4) Fetch the connection stats from the system probe
    ```
    curl -s --unix-socket /opt/datadog-agent/run/sysprobe.sock http://unix/connections?client\_id=1 | jq .
    ```
5) Find the connection stats corresponding to the http requests you made & verify that they were recorded in the "httpStatsByPath" field
